### PR TITLE
test(waku-filter): Subscribe tests batch (3/4)

### PR DESCRIPTION
**Pull request by AlejandroCabeza**
# Description
Implement the third batch of subscribe tests for waku filter, finishing the client part.

# Changes
- [X] Finish implementing waku filter client subscribe tests.
- [X] Fix MaxCriteriaPerSubscription usage, so it actually allows up to (including) the number defined in it.

# Dependencies
https://github.com/waku-org/nwaku/pull/2035

### CODE REVIEW

File: tests/waku_filter_v2/test_waku_client.nim [hunk 1]
```diff
@@ -634,54 +646,27 @@ suite "Waku Filter - End to End":
...
       # TODO: MaxCriteriaPerSubscription comparison
       # TODO: MaxCriteriaPerSubscription behaviour when exceeding limit and topics are contained
-      xasyncTest "Max Criteria Per Subscription":  
+      asyncTest "Max Criteria Per Subscription":  
         # Given a topic list of size MaxCriteriaPerSubscription
         var topicSeq: seq[string] = toSeq(0..<MaxCriteriaPerSubscription).mapIt("topic" & $it)
-        
-        # When client service node subscribes to the topic list of size MaxCriteriaPerSubscription
+      
+        # When client serice node subscribes to the topic list of size MaxCriteriaPerSubscription
```
Comment (SionoiS) [inline tests/waku_filter_v2/test_waku_client.nim:653]:
```suggestion
        # When client service node subscribes to the topic list of size MaxCriteriaPerSubscription
```

File: tests/waku_filter_v2/test_waku_client.nim [hunk 2]
```diff
@@ -634,54 +646,27 @@ suite "Waku Filter - End to End":
       
       # TODO: MaxCriteriaPerSubscription comparison
       # TODO: MaxCriteriaPerSubscription behaviour when exceeding limit and topics are contained
```
Comment (SionoiS) [inline tests/waku_filter_v2/test_waku_client.nim:648]: Should we remove this?

**Review by SionoiS — APPROVED** [review]
LGTM

Thanks!

File: tests/waku_filter_v2/test_waku_client.nim (hunk 2 above)
Comment (AlejandroCabeza) [reply tests/waku_filter_v2/test_waku_client.nim:648]: Indeed! Removed.

File: tests/waku_filter_v2/test_waku_client.nim (hunk 1 above)
Comment (AlejandroCabeza) [reply tests/waku_filter_v2/test_waku_client.nim:653]: Fixed

**Review by Ivansete-status — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**